### PR TITLE
Add support for sticky list headers

### PR DESCRIPTION
**Pull request by kylebshr**
Adds `stickyListHeader`, allowing you to pin the list header to the top of the visible frame as you scroll.

- [x] Add tests

https://user-images.githubusercontent.com/3526783/179046891-8808eee3-efb4-4f40-a5d4-6d2cba89f1f8.mp4

### Checklist

Please do the following before merging:

- [x] Ensure any public-facing changes are reflected in the [changelog](https://github.com/kyleve/Listable/blob/main/CHANGELOG.md). Include them in the `Main` section.

### CODE REVIEW

File: ListableUI/Sources/Layout/Flow/FlowListLayout.swift [hunk 1]
```diff
@@ -105,6 +105,8 @@ public struct FlowAppearance : ListLayoutAppearance {
     
     /// The direction the flow layout will be laid out in.
     public var direction: LayoutDirection
+
+    public var stickyListHeader: Bool
```
Comment (kylebshr) [inline ListableUI/Sources/Layout/Flow/FlowListLayout.swift:109]: - [ ]Doc me

File: ListableUI/Sources/Layout/Table/TableListLayout.swift [hunk 2]
```diff
@@ -108,6 +108,8 @@ public struct TableAppearance : ListLayoutAppearance
     
     /// How the layout should flow, either horizontally or vertically.
     public var direction: LayoutDirection
+
+    public var stickyListHeader: Bool
```
Comment (kylebshr) [inline ListableUI/Sources/Layout/Table/TableListLayout.swift:112]: - [ ] Doc me

File: ListableUI/Sources/Layout/ListLayout/ListLayout.swift [hunk 3]
```diff
@@ -199,14 +205,54 @@ extension AnyListLayout
...
+        let visibleContentFrame = self.visibleContentFrame(for: collectionView)
+
+        let header = self.content.header
+
+        let headerOrigin = self.direction.y(for: header.defaultFrame.origin)
+        let visibleContentOrigin = self.direction.y(for: visibleContentFrame.origin)
+
+        if headerOrigin < visibleContentOrigin {
+
+            // Make sure the pinned origin stays within the section's frame.
```
Comment (kyleve) [inline ListableUI/Sources/Layout/ListLayout/ListLayout.swift:222]: Is `section` relevant here given this isn't in a section?

Comment (kylebshr) [reply ListableUI/Sources/Layout/ListLayout/ListLayout.swift:222]: oops copy-pasted, no

File: ListableUI/Tests/Layout/Table/TableListLayoutTests.swift [hunk 4]
```diff
@@ -77,6 +77,17 @@ class TableListLayoutTests : XCTestCase
         snapshot.test(output: LayoutAttributesSnapshot.self)
     }
 
+    func test_layout_vertical_stickyHeader()
```
Comment (kyleve) [inline ListableUI/Tests/Layout/Table/TableListLayoutTests.swift:80]: Is this right? It looks like there's overlap; which seems wrong.

image

Comment (kylebshr) [reply ListableUI/Tests/Layout/Table/TableListLayoutTests.swift:80]: Unless one of the squares is a section header?

Comment (kyleve) [reply ListableUI/Tests/Layout/Table/TableListLayoutTests.swift:80]: I am not 100% sure – hence why I am asking!

**Review by kyleve — APPROVED** [review]

File: ListableUI/Tests/Layout/Table/TableListLayoutTests.swift (hunk 4 above)
Comment (kylebshr) [reply ListableUI/Tests/Layout/Table/TableListLayoutTests.swift:80]: It's a footer, so it's correct, but I updated it to test that the sticky section header works correctly. Also gave footers a different color than headers.

Comment (kyleve) [reply ListableUI/Tests/Layout/Table/TableListLayoutTests.swift:80]: Thank you!

**Review by kyleve — APPROVED** [review]

**Pull request merged**